engine/events/weather.asm

```diff
@@ -0,0 +1,195 @@
+MACRO ow_weather_map
+	map_id \1
+	dw \2 ; event
+	shift
+	rept NUM_OW_WEATHERS
+		shift
+		db (\1) percent
+	endr
+ENDM
+
+SetCurrentWeather::
+	farcall GetOvercastIndex
+	and a
+	jr z, .not_overcast
+	call Random
+	cp 25 percent
+	; a = carry ? OW_WEATHER_THUNDERSTORM : OW_WEATHER_RAIN
+	sbc a
+	and OW_WEATHER_THUNDERSTORM - OW_WEATHER_RAIN
+	add OW_WEATHER_RAIN
+.set_weather
+	ld b, a
+	ld a, [wWeatherFlags]
+	bit OW_WEATHER_DO_FLY_F, a
+	jr nz, .skip_cooldown
+	; if there was weather running then no cooldown
+	ld a, [wCurWeather]
+	and a
+	jr z, .skip_cooldown
+	; No cooldown if switching from rain to thunderstorm or vice versa
+	ld c, a
+	cp OW_WEATHER_RAIN
+	jr nz, .not_rain
+	ld a, b
+	cp OW_WEATHER_THUNDERSTORM
+	jr z, .skip_cooldown
+.not_rain
+	ld a, c
+	cp OW_WEATHER_THUNDERSTORM
+	jr nz, .compare_cur_and_prev_weather
+	ld a, b
+	cp OW_WEATHER_RAIN
+	jr z, .skip_cooldown
+	ld a, c
+.compare_cur_and_prev_weather
+	; if weather is different from previous weather then do cooldown
+	cp b
+	jr nz, .do_cooldown
+.skip_cooldown
+	ld a, b
+	ld [wCurWeather], a
+	farcall LoadWeatherPal
+	call LoadWeatherGraphics
+	xor a
+	ld [wOverworldWeatherCooldown], a
+	ret
+
+.do_cooldown
+	ld [wPrevWeather], a
+	ld a, b
+	ld [wCurWeather], a
+	ld a, 32
+	ld [wOverworldWeatherCooldown], a
+	ret
+
+.not_overcast
+	ld a, [wMapGroup]
+	cp GROUP_SNOWTOP_MOUNTAIN_OUTSIDE
+	jr nz, .not_snowing
+	ld a, [wMapNumber]
+	cp MAP_SNOWTOP_MOUNTAIN_OUTSIDE
+	jr z, .snowing
+	cp MAP_SNOWTOP_MOUNTAIN_INSIDE
+	jr nz, .not_snowing
+.snowing
+	ld a, OW_WEATHER_SNOW
+	jr .set_weather
+
+.not_snowing
+	ld a, [wMapGroup]
+	cp GROUP_RUGGED_ROAD_NORTH
+	jr nz, .no_weather
+	ld a, [wMapNumber]
+	cp MAP_RUGGED_ROAD_NORTH
+	jr z, .sandstorm
+	cp MAP_RUGGED_ROAD_SOUTH
+	jr nz, .no_weather
+.sandstorm
+	ld a, OW_WEATHER_SANDSTORM
+	jr .set_weather
+
+.no_weather
+	ld a, OW_WEATHER_NONE
+	jr .set_weather
+
+
+;OverworldWeatherMaps:
```

Comment:
Is this commented code+data unfinished? obsolete? moved elsewhere?

Reply:
This was the original implementation I was going with for CSE... I originally left it in to should you an alternative way to set the weather based on probabilities, events, locations, ect... I forgot to mention it and its not needed anymore.. for now.